sentry/src/main/java/io/sentry/hints/Hints.java

```diff
@@ -9,21 +13,51 @@ public final class Hints {
 
   private final @NotNull Map<String, Object> internalStorage = new HashMap<String, Object>();
 
+  public static @NotNull Hints withAttachment(@Nullable Attachment attachment) {
+    @NotNull final Hints hints = new Hints();
+    hints.getAttachmentContainer().add(attachment);
+    return hints;
+  }
+
+  public static @NotNull Hints withAttachments(@Nullable List<Attachment> attachments) {
+    @NotNull final Hints hints = new Hints();
+    hints.getAttachmentContainer().addAll(attachments);
+    return hints;
+  }
+
   public void set(@NotNull String hintType, @Nullable Object hint) {
     internalStorage.put(hintType, hint);
   }
 
-  public @Nullable Object get(@NotNull String hintType) {
-    return internalStorage.get(hintType);
+  public @Nullable Object get(@NotNull String hintName) {
+    return internalStorage.get(hintName);
   }
 
-  // TODO maybe not public
-  public void remove(@NotNull String hintType) {
-    internalStorage.remove(hintType);
+  @SuppressWarnings("unchecked")
+  public <T extends Object> @Nullable T getAs(@NotNull String hintName, @NotNull Class<T> clazz) {
+    Object hintValue = internalStorage.get(hintName);
+
+    if (clazz.isInstance(hintValue)) {
+      return (T) hintValue;
+    } else {
+      return null;
+    }
+  }
+
+  public void remove(@NotNull String hintName) {
+    internalStorage.remove(hintName);
   }
 
-  // TODO addAttachment(one)
-  // TODO getAttachments(): List
-  // TODO setAttachments(list)
-  // TODO clearAttachments()
+  public @NotNull AttachmentContainer getAttachmentContainer() {
```

Comment:
`m`: This would make the API slimmer.
```suggestion
  public @NotNull AttachmentContainer getAttachments() {
```